### ISSUE TITLE
Move to new startDebugging API - vscode.startDebug deprecated now

### ISSUE THREAD

**Issue by moskal91** [Unity-Technologies/vscode-unity-debug#54]
Since VS Code 1.16, `vscode.startDebug` is deprecated:
https://github.com/Microsoft/vscode/issues/33795

**Comment by moskal91** [comment]
@lukaszunity Update: In the latest VS Code 1.18, "Unity Attach Debugger" command does nothing and returns error in console:
```
[Extension Host] Error: command 'vscode.startDebug' not found
    at t._tryExecuteCommand (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:1117177)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:1117016
    at n.Class.define.cancel.then (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:69294)
    at t.executeCommand (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:1116990)
    at e.$executeCommand (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2822547)
    at t.e.invoke (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2876274)
    at e._invokeHandler (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2440074)
    at e._receiveOneMessage (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2439764)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2438791
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:9:2440710
```

**Comment by ahuertabhg** [comment]
I don't think this can be considered an enhancement any more as it's blocking everyone from upgrading beyond 1.17.

**Comment by lukaszunity** [comment]
I'm unable to reproduce the error in the console with VS Code 1.18.1 on macOS when using "Unity Attach Debugger"

**Comment by lukaszunity** [comment]
Turns out I forgot to actually attach and then I got the error in the console :)

**Comment by lukaszunity** [comment]
Fixed in 501a74534b1594ab5af2b31adcee81fec90669dc. 

Turns out there is a startDebugging bug (https://github.com/Microsoft/vscode/issues/37559) in VS Code 1.18.1 that makes the fix not work correctly. Works correctly in VS Code Insiders, so I'm going to wait with publishing a new version of the debugger extension until the startDebugging issue has been fixed in the public version.

**Comment by bumpmann** [comment]
I m on vscode 1.19.1 and i don't have this error but this is still marked as a duplicate of #65, any news about this ?

**Comment by lukaszunity** [comment]
Fixed in published 1.3.0 version.
